modules/rosetta/src/main/scala/org/tessellation/rosetta/domain/Amount.scala

```diff
@@ -1,31 +1,60 @@
 package org.tessellation.rosetta.domain
 
 import org.tessellation.ext.derevo.magnoliaCustomizable.snakeCaseConfiguration
-import org.tessellation.rosetta.domain.currency.Currency
+import org.tessellation.rosetta.domain.currency.{Currency, DAG}
+import org.tessellation.schema.transaction.{TransactionAmount, TransactionFee}
 
-import derevo.cats.eqv
+import derevo.cats.{eqv, show}
 import derevo.circe.magnolia._
 import derevo.derive
 import eu.timepit.refined.api.Refined
 import eu.timepit.refined.auto._
 import eu.timepit.refined.boolean.Not
 import eu.timepit.refined.cats._
 import eu.timepit.refined.generic.Equal
+import eu.timepit.refined.numeric.{NonNegative, Positive}
+import eu.timepit.refined.refineV
 import io.circe.refined._
 import io.estatico.newtype.macros.newtype
 
+// TODO should we rename Amount to NonZeroAmount ?
+//  it would be clearer and not clash with schema.balance.Amount
```

Comment:
I think we can rename `Amount` to `NonZeroAmount`.
@Constellation-Labs/protocol wdyt?

Reply:
@sdavidp also resolved without comment, unresolving

Reply:
@marcinwadon I resolved it because it seemed like the conversation had stopped. Since the rename was a suggestion on my part I didn't think it was a show-stopper to resolve it, especially since we can unresolve if want to continue discussing it.